### DISCUSSION TITLE
Warn users when debugging optimized binaries

### DISCUSSION THREAD

**Discussion by JadenSimon** [golang/vscode-go#1446] [Discussions]
Delve allows users to debug optimized binaries, however, when using the Delve client it will warn you that the function is optimized after hitting a breakpoint. This means the user is not confused when local variables are not what they expect (i.e. missing or invalid). The Go extension's DA makes no attempt at alerting the user that they might be debugging an optimized binary, causing a lot of confusion when debugging does not work as expected. This is especially true when remote debugging, as it may be up to an external process to do the compilation.

**Comment by hyangah** [comment]
Sorry that I didn't see this earlier. Good idea. Not sure what is the best way to reflect this in the UI though.
Filed https://github.com/golang/vscode-go/issues/1471 for discussion.

**Reply by JadenSimon** [reply]
No worries. Glad to see it's being considered!

**Reply by hyangah** [reply]
This feature is now available in the latest dlv (built at master). You can see it when using the dlv-dap mode.
(https://github.com/golang/vscode-go/blob/master/docs/dlv-dap.md)
